src/qforte/utils/qbt_tapering.py

```diff
@@ -0,0 +1,321 @@
+    bottom = np.concatenate((np.identity(n_qubits, dtype=np.uint8), np.zeros((n_qubits, n_qubits), dtype=np.uint8)), axis=1)
+    blnr_frm_mtrx = np.concatenate((top, bottom), axis=0)
+
+    ## Construct abln matrix.
+    ## abln[i,j] contains the value of the bilinear form betweeen the ith and jth basis vectors of ker{E}
+    ## A "0"/"1" in the ith row and jth column of abln means that the Pauli strings
+    ## associated with the ith and jth binary vectors commute/anticommute.
+
+    aux = np.matmul(blnr_frm_mtrx, basis.T)
+
+    abln = np.zeros((basis.shape[0], basis.shape[0]), dtype=np.uint8)
+    for row in range(basis.shape[0]):
+        for column in range(basis.shape[0]):
+            for idx in range(2*n_qubits):
+                abln[row, column] = (abln[row, column] + basis[row, idx] * aux[idx, column]) % 2
+
+    ## Find the maximal abelian subgroup iteratively. This is done as follows:
+    ## 1) Find non-zero row of abln with smallest Hamming weight.
+    ##    The corresponding basis vector e commutes with most other basis vectors.
+    ## 2) Find basis vectors that do not commute with e and remove them.
+    ## 3) Reconstruct abln matrix with remaining basis vectors.
+    ## 4) If abln == 0 exit otherwise go to step 1) and repeat.
+
+    basis_tmp = basis
+
+    while not np.all(abln == 0):
+
+        ### Maximum possible Hamming weight of a given binary vector
+        hmng = 2 * n_qubits
+
+        idx = 0
+        for row in range(abln.shape[0]):
+            ### Find the basis vector e that commutes with most of the remaining basis vectors
+            hmng_tmp = np.sum(abln[row, :])
+            if hmng_tmp > 0 and hmng > hmng_tmp:
+                hmng = hmng_tmp
+                idx = row
+
+        ### Identify basis vectors that do not commute with e and eliminate them.
+        nonzero_indices = np.argwhere(abln[idx,:])
+        basis_tmp = np.delete(basis_tmp, nonzero_indices, axis=0)
+
+        ### Construct abln matrix for the next iteration.
+
+        aux = np.matmul(blnr_frm_mtrx, basis_tmp.T)
+
+        abln = np.zeros((basis_tmp.shape[0], basis_tmp.shape[0]), dtype=np.uint8)
+        for row in range(basis_tmp.shape[0]):
+            for column in range(basis_tmp.shape[0]):
+                for idx in range(2*n_qubits):
+                    abln[row, column] = (abln[row, column] + basis_tmp[row, idx] * aux[idx, column]) % 2
+
+    tau = basis_tmp
+
+    #### This step is not necessary. It transforms the generators of the symmetry group in such a way that the
+    #### removed qubits have the smallest/largest possbile identity numbers. A similar modification is made
+    #### at the "sigma_x selection" part of the algorithm below. The "smallest" case produces identical resutls
+    #### with those obtained using Qiskit, after taking into account that in Qiskit Slater determinants are
+    #### represented by strings of alpha orbitals followed by the beta ones.
```

Comment:
```suggestion
    #### at the "sigma_x selection" part of the algorithm below.
```

We don't want our comment to be invalidated, should qiskit ever change their code.